src/span.js

```diff
@@ -18,6 +100,24 @@ export default class Span {
     // ---------------------------------------------------------------------- //
 
     /**
+     * Returns the SpanContext object associated with this Span.
+     *
+     * @return {SpanContext}
+     */
+    context() {
+        if (API_CONFORMANCE_CHECKS) {
+            if (arguments.length !== 0) {
+                throw new Error('Invalid number of arguments');
+            }
+        }
+        if (this._imp) {
+            return new Tracer(this._imp.context());
+        }
+        // XXX: what to do here?
```

Comment:
* I don't think there should be null checks anywhere except maybe in constructors
* the API wrapper objects should be created eagerly, e.g. when `Span(SpanImpl())` is constructed, it immediately captures `Context(impl.context())`, instead of creating a new one every time